### ISSUE TITLE
Mismatch in simulation by using saveat in fmi2SimulateME

### ISSUE THREAD

**Issue by sathvikbhagavan** [ThummeTo/FMI.jl#120]
FMU used: https://github.com/modelica/fmi-cross-check/blob/master/fmus/2.0/me/linux64/MapleSim/2021.2/CoupledClutches/CoupledClutches.fmu

Code to reproduce this error:

```
using FMI, Test, DifferentialEquations

fmu_coupled_clutches_ME = FMI.fmi2Load(joinpath(@__DIR__, "CoupledClutches.fmu"), type=:ME)
outputs = FMI.FMIImport.fmi2GetOutputNames(fmu_coupled_clutches_ME) .|> string

sim1 = FMI.fmi2SimulateME(fmu_coupled_clutches_ME;
                        recordValues=outputs,
                        solver=Tsit5(),
                        saveat=[i for i in 0:0.01:1.5]);

sim2 = FMI.fmi2SimulateME(fmu_coupled_clutches_ME;
                        recordValues=nothing,
                        solver=Tsit5(),
                        saveat=[i for i in 0:0.01:1.5]);

@test sim1.values !== nothing
@test sim2.values === nothing
@test isapprox(sim1.states.t, sim2.states.t)

## This is erroring out (but it should not)
@test isapprox(sim1.states.u[5], sim2.states.u[5])
## 

sim3 = FMI.fmi2SimulateME(fmu_coupled_clutches_ME;
                        recordValues=outputs,
                        solver=Tsit5());

sim4 = FMI.fmi2SimulateME(fmu_coupled_clutches_ME;
                        recordValues=nothing,
                        solver=Tsit5());

@test sim3.values !== nothing
@test sim4.values === nothing
@test isapprox(sim3.states.t, sim4.states.t)
@test isapprox(sim3.states.u, sim4.states.u)
```

```
[0c46a032] DifferentialEquations v7.1.0
[14a09403] FMI v0.8.5
[8dfed614] Test
```

The states in sim1 and sim2 should match as the only difference is in the recordValues argument which just stores values.
Also, without using saveat, the states in sim3 and sim4 are matching.

@ThummeTo

cc: @sharanry

**Comment by ThummeTo** [comment]
Yes, I can reproduce:
```
using Statistics
mean(abs.(sim1.states.u[5]-sim2.states.u[5])) = 0.0017112668432117551
max(abs.(sim1.states.u[5]-sim2.states.u[5])...) = 0.03042616503656248
```

You can also try:
```
fig = fmiPlot(sim1; stateIndices=5:5, values=false)
fmiPlot!(fig, sim2; stateIndices=5:5, values=false)
```

Question is, if this is an implementation error or just a numerical precision thing... in fact the solver makes little breakes at the `saveat`-timepoints to retrieve FMU-values, so even if theses are not "accepted solver steps", there is a look-up at these positions.

I will check this out ...

**Comment by ThummeTo** [comment]
EDIT: The timepoints of the events are also sligthly different ...

**Comment by ThummeTo** [comment]
this should be fixed in v0.8.6 :-)
